### PR TITLE
Simplify gem-handling

### PR DESCRIPTION
**Pull request by sandstrom**
1. Calculate the list of files dynamically (easier than manual updates). Same method as the [official bundler gemspec template file](https://github.com/bundler/bundler/blob/master/lib/bundler/templates/newgem/newgem.gemspec.tt) use.

2. Delete `Gemfile.lock` and point `Gemfile` to `ruby-vips.gemspec`.
  
  This is the canonical way of handling this stuff:
  http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/

### CODE REVIEW

File: ruby-vips.gemspec [hunk 1]
```diff
@@ -19,60 +19,9 @@ Gem::Specification.new do |s|
...
-    "spec/samples/lcd.icc",
-    "spec/samples/lion.svg",
-    "spec/samples/sample.csv",
-    "spec/samples/sample.exr",
-    "spec/samples/wagon.jpg",
-    "spec/samples/wagon.v",
-    "spec/spec_helper.rb",
-    "spec/vips_spec.rb"
-  ]
+  s.files = `git ls-files -z`.split("\x0").reject do |f|
```
Comment (jcupitt) [inline ruby-vips.gemspec]: ruby-vips.gemspec is generated by jeweler, I think this change will be lost

File: Gemfile [hunk 2]
```diff
@@ -1,15 +1,4 @@
...
-group :development do
-    gem "rspec", "~> 3.3"
-    gem "yard", "~> 0.8"
-    gem "redcarpet", "~> 3.3"
-    gem "github-markup", "~> 1.4"
-    gem "bundler", "~> 1.0"
-    gem "jeweler", "~> 2.0"
-end
+# all gems are handled by .gemspec (since this repo is for a gem)
+gemspec
```
Comment (jcupitt) [inline Gemfile]: does it matter that we don't list development dependencies like yard?

**Comment by jcupitt** [comment]
Thanks! It's great to have all this stuff fixed, I'm not much of a ruby programmer. 

I wonder about the gemspec though, isn't jeweler generating this?

File: Gemfile (hunk 2 above)
Comment (sandstrom) [inline Gemfile]: By deferring to `gemspec` it'll use the ones outlined here:
https://github.com/sandstrom/ruby-vips-1/blob/d740709ce58369af93e9d07f8d8eb912004f0567/ruby-vips.gemspec#L34-L40

File: ruby-vips.gemspec (hunk 1 above)
Comment (sandstrom) [inline ruby-vips.gemspec]: Possibly! I haven't used jeweler, don't know much about it.

For what tasks do you currently use jeweler?

**Comment by sandstrom** [comment]
Possibly, I haven't used jeweler, don't know much about it.

For what tasks do you currently use jeweler?

**Comment by jcupitt** [comment]
Jeweler makes the gem and handles all the interactions with ruby-gems.org. You have a small block in Rakefile and it does everything from that. 

I think @stanislaw added the jeweler dependancy. Perhaps it's less useful than it used to be?

**Comment by stanislaw** [comment]
No input from me as I didn't work with Ruby, gems, jeweler for quite a while.

My guess would be that, yes, these days jeweler is less useful.

**Comment by jcupitt** [comment]
Thanks @stanislaw, sorry to ping you.

**Comment by jcupitt** [comment]
Googling, it seems jeweler has been abandoned. 

https://github.com/technicalpickles/jeweler/issues/283

I suppose we should probably switch to vanilla ruby. @sandstrom, can I close this pull request and make an issue for removing jeweler instead?

**Comment by sandstrom** [comment]
I found this tutorial, my guess is that gem-handling has gotten much easier since the heydays of jeweler.

https://www.digitalocean.com/community/tutorials/how-to-package-and-distribute-ruby-applications-as-a-gem-using-rubygems

*** 

@jcupitt This isn't really concerned about jeweler, i.e. even if jeweler is removed (which I have nothing against) I still suggest making these changes.

**Comment by jcupitt** [comment]
OK, I'll remove jeweler separately. Thanks @sandstrom and stanislaw!

**Comment by jcupitt** [comment]
This branch seems to work without jeweler:

https://github.com/jcupitt/ruby-vips/tree/remove-jeweler

I'll merge to master in a few days.

**Comment by sandstrom** [comment]
@jcupitt Nice! 😄

**Pull request merged**